substrate/client/consensus/beefy/src/keystore.rs

```diff
@@ -71,54 +88,137 @@ impl BeefyKeystore {
 	/// Note that `message` usually will be pre-hashed before being signed.
 	///
 	/// Return the message signature or an error in case of failure.
-	pub fn sign(&self, public: &Public, message: &[u8]) -> Result<Signature, error::Error> {
+	pub fn sign(
+		&self,
+		public: &AuthorityId,
+		message: &[u8],
+	) -> Result<<AuthorityId as RuntimeAppPublic>::Signature, error::Error> {
 		let store = self.0.clone().ok_or_else(|| error::Error::Keystore("no Keystore".into()))?;
 
-		let msg = keccak_256(message);
-		let public = public.as_ref();
-
-		let sig = store
-			.ecdsa_sign_prehashed(BEEFY_KEY_TYPE, public, &msg)
-			.map_err(|e| error::Error::Keystore(e.to_string()))?
-			.ok_or_else(|| error::Error::Signature("ecdsa_sign_prehashed() failed".to_string()))?;
-
-		// check that `sig` has the expected result type
-		let sig = sig.clone().try_into().map_err(|_| {
-			error::Error::Signature(format!("invalid signature {:?} for key {:?}", sig, public))
+		//ECDSA shoud do ecdsa_sign_prehashed because it needs to be hashed by keccak_256
+		//as such we need to deal with signing case by case
+
```

Comment:
```suggestion

		// ECDSA should use ecdsa_sign_prehashed since it needs to be hashed by keccak_256 instead of blake2.
		// As such we need to deal with producing the signatures case-by-case
```